src/viam/sdk/components/motor/client.cpp

```diff
@@ -141,66 +81,27 @@ grpc::StatusCode MotorClient::stop(const AttributeMap& extra) {
 }
 
 Motor::power_status MotorClient::get_power_status(const AttributeMap& extra) {
-    viam::component::motor::v1::IsPoweredRequest request;
-    viam::component::motor::v1::IsPoweredResponse response;
-
-    grpc::ClientContext ctx;
-    set_client_ctx_authority(ctx);
-
-    *request.mutable_name() = this->name();
-    *request.mutable_extra() = map_to_struct(extra);
-
-    const grpc::Status status = stub_->IsPowered(&ctx, request, &response);
-    if (!status.ok()) {
-        throw std::runtime_error(status.error_message());
-    }
-    return from_proto(response);
+    return make_client_helper(this, *stub_, &StubType::IsPowered)
+        .with(extra)
+        .invoke([](auto& response) { return from_proto(response); });
 }
 
 std::vector<GeometryConfig> MotorClient::get_geometries(const AttributeMap& extra) {
-    viam::common::v1::GetGeometriesRequest req;
-    viam::common::v1::GetGeometriesResponse resp;
-    grpc::ClientContext ctx;
-    set_client_ctx_authority(ctx);
-
-    *req.mutable_name() = this->name();
-    *req.mutable_extra() = map_to_struct(extra);
-
-    stub_->GetGeometries(&ctx, req, &resp);
-    return GeometryConfig::from_proto(resp);
-};
+    return make_client_helper(this, *stub_, &StubType::GetGeometries)
+        .with(extra)
+        .invoke([](auto& response) { return GeometryConfig::from_proto(response); });
+}
 
 bool MotorClient::is_moving() {
-    viam::component::motor::v1::IsMovingRequest request;
-    viam::component::motor::v1::IsMovingResponse response;
-
-    grpc::ClientContext ctx;
-    set_client_ctx_authority(ctx);
-
-    *request.mutable_name() = this->name();
-
-    const grpc::Status status = stub_->IsMoving(&ctx, request, &response);
-    if (!status.ok()) {
-        throw std::runtime_error(status.error_message());
-    }
-    return response.is_moving();
+    return make_client_helper(this, *stub_, &StubType::IsMoving).invoke([](auto& response) {
```

Comment:
Note the lack of a `.with` call here: there are no params and no `extra` so we don't have anything to prepare.